src/structs.rs

```diff
@@ -68,6 +68,7 @@ pub enum SwimBroadcast {
 pub enum SwimMessage {
     Ping,
     PingRequest(Peer),
-    Ack(Peer),
+    Ack(Peer, String, u32),
```

Comment:
Almost to the point where these should be named fields, but I'm holding off for now. (If any of them had the same type, I'd have refactored by now.)